packages/office-ui-fabric-react/src/components/pickers/AutoFill/BaseAutoFill.tsx

```diff
@@ -154,37 +167,71 @@ export class BaseAutoFill extends BaseComponent<IBaseAutoFillProps, IBaseAutoFil
       this.props.onKeyDown(ev);
     }
 
-    switch (ev.which) {
-      case KeyCodes.backspace:
-        this._autoFillEnabled = false;
-        break;
-      case KeyCodes.left:
-        if (this._autoFillEnabled) {
+    // If the event is actively being composed, then don't alert autofill.
+    // Right now typing does not have isComposing, once that has been fixed any should be removed.
+    if (!(ev.nativeEvent as any).isComposing) {
+      switch (ev.which) {
+        case KeyCodes.backspace:
           this._autoFillEnabled = false;
-        }
-        break;
-      case KeyCodes.right:
-        if (this._autoFillEnabled) {
-          this._autoFillEnabled = false;
-        }
-        break;
-      default:
-        if (!this._autoFillEnabled) {
-          if (this.props.enableAutoFillOnKeyPress!.indexOf(ev.which) !== -1) {
-            this._autoFillEnabled = true;
+          break;
+        case KeyCodes.left:
+          if (this._autoFillEnabled) {
+            this._value = this.state.displayValue!;
+            this._autoFillEnabled = false;
           }
-        }
-        break;
+          break;
+        case KeyCodes.right:
+          if (this._autoFillEnabled) {
+            this._value = this.state.displayValue!;
+            this._autoFillEnabled = false;
+          }
+          break;
+        default:
+          if (!this._autoFillEnabled) {
+            if (this.props.enableAutoFillOnKeyPress!.indexOf(ev.which) !== -1) {
+              this._autoFillEnabled = true;
+            }
+          }
+          break;
+      }
     }
   }
 
   @autobind
   private _onChange(ev: React.FormEvent<HTMLElement>) {
-    let value: string = (ev.target as HTMLInputElement).value;
-    if (value && (ev.target as HTMLInputElement).selectionStart === value.length && !this._autoFillEnabled && value.length > this._value.length) {
+    let value: string = this._getCurrentInputValue(ev);
+    // Right now typing does not have isComposing, once that has been fixed any should be removed.
+    this._tryEnableAutoFill(value, this._value, (ev.nativeEvent as any).isComposing);
+    this._updateValue(value);
+  }
+
+  private _getCurrentInputValue(ev?: React.FormEvent<HTMLElement>) {
+    if (ev && ev.target && (ev.target as any).value) {
+      return (ev.target as any).value;
+    }
+    let value: string = this._inputElement.value;
+    return value;
+  }
+
+  /**
+   * Attempts to enable autofill. Whether or not autofill is enabled depends on the input value,
+   * whether or not any text is selected, and only if the new input value is longer than the old input value.
+   * isComposed is whether or not the new value is a composed value.
+   * Autofill should never be set to true if the value is composing. Once compositionEnd is called, then
```

Comment:
"value is BEING composed"? not sure of the right verbiage here...
also you should describe what composition is